dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/AdminDataManipulationApi.kt

```diff
@@ -0,0 +1,45 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import jakarta.validation.Valid
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+
+/**
+ * Defines the restful dataland-backend API regarding deletion endpoint for admin users
+ */
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+@RequestMapping("/data/{dataId}")
+fun interface AdminDataManipulationApi {
+
+    /**
+     * A method to delete data via Dataland into a data store
+     * @param dataId consisting of the ID of the dataset to be removed
+     * @return meta info about the stored data including the ID of the created entry in the data store
+     */
+    @Operation(
+        summary = "Delete a data set.",
+        description = "The data is removed from the data store.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully deleted the dataset."),
+            ApiResponse(responseCode = "404", description = "Dataset does not exist."),
```

Comment:
@stewal19 should we add here 401 (unauthorized) and 403 (client error) as well?

Reply:
In my opinion we should do it the same way, as we did in the companies @head endpoint. There we also did not specify the 401 and 403 errors. Maybe check out the companies head endpoint and see how it is implemented there and if it is then fine for you we can keep those two errors out from here.